src/Hub.cpp

```diff
@@ -38,24 +38,28 @@ bool SortByDistance::operator()(const Expansion& lhs_, const Expansion& rhs_) co
 Hub::Hub(sc2::Race current_race_, const Expansions& expansions_):
     m_current_race(current_race_), m_expansions(expansions_),
     m_current_supply_type(sc2::UNIT_TYPEID::INVALID),
-    m_current_worker_type(sc2::UNIT_TYPEID::INVALID) {
+    m_current_worker_type(sc2::UNIT_TYPEID::INVALID),
+    m_current_townhall_type(sc2::UNIT_TYPEID::INVALID) {
     std::sort(m_expansions.begin(), m_expansions.end(),
         SortByDistance(gAPI->observer().StartingLocation()));
 
     switch (m_current_race) {
         case sc2::Race::Protoss:
             m_current_supply_type = sc2::UNIT_TYPEID::PROTOSS_PYLON;
             m_current_worker_type = sc2::UNIT_TYPEID::PROTOSS_PROBE;
+            m_current_townhall_type = sc2::UNIT_TYPEID::PROTOSS_NEXUS;
```

Comment:
This looks a bit confusing, because this is not a townhall type, but a kind of 'new expansion building' type. TownHall for terrans and protoss has much more types and IDs.

Reply:
"m_current_new_expansion_building_type" is a bit long, but adding 'new' should be more specific enough: m_current_new_townhall_type.

Reply:
Postponed until after Strategy Settings is done which can specific macro settings.